Mathlib/CategoryTheory/Comma/Presheaf.lean

```diff
@@ -0,0 +1,575 @@
+lemma app_val {F : Cᵒᵖ ⥤ Type v} {η : F ⟶ A} {X : C} {s : yoneda.obj X ⟶ A}
+    (p : OverArrows η s) : η.app (op X) p.val = yonedaEquiv s :=
+  p.prop.app
+
+/-- In the special case `F = yoneda.obj Y`, the element `p.val` for `p : OverArrows η s` is itself
+    a morphism `X ⟶ Y`. -/
+@[simp]
+lemma map_val {Y : C} {η : yoneda.obj Y ⟶ A} {X : C} {s : yoneda.obj X ⟶ A}
+    (p : OverArrows η s) : yoneda.map p.val ≫ η = s := by
+  rw [← yonedaEquiv.injective.eq_iff, yonedaEquiv_comp, yonedaEquiv_yoneda_map]
+  simp only [unop_op, p.app_val]
+
+/-- Functoriality of `OverArrows η s` in `η`. -/
+def map₁ {F G : Cᵒᵖ ⥤ Type v} {η : F ⟶ A} {μ : G ⟶ A} {X : C} {s : yoneda.obj X ⟶ A}
+    (u : OverArrows η s) (ε : F ⟶ G) (hε : ε ≫ μ = η) : OverArrows μ s :=
+  ⟨ε.app _ u.val, MakesOverArrow.map₁ hε u.2⟩
+
+@[simp]
+lemma map₁_val {F G : Cᵒᵖ ⥤ Type v} {η : F ⟶ A} {μ : G ⟶ A} {X : C}
+    (s : yoneda.obj X ⟶ A) (u : OverArrows η s) (ε : F ⟶ G) (hε : ε ≫ μ = η) :
+    (u.map₁ ε hε).val = ε.app _ u.val :=
+  rfl
+
+/-- Functoriality of `OverArrows η s` in `s`. -/
+def map₂ {F : Cᵒᵖ ⥤ Type v} {η : F ⟶ A} {X Y : C} {s : yoneda.obj X ⟶ A}
+    {t : yoneda.obj Y ⟶ A} (u : OverArrows η t) (f : X ⟶ Y) (hst : yoneda.map f ≫ t = s) :
+    OverArrows η s :=
+  ⟨F.map f.op u.val, MakesOverArrow.map₂ f hst u.2⟩
+
+@[simp]
+lemma map₂_val {F : Cᵒᵖ ⥤ Type v} {η : F ⟶ A} {X Y : C} (f : X ⟶ Y)
+    {s : yoneda.obj X ⟶ A} {t : yoneda.obj Y ⟶ A} (hst : yoneda.map f ≫ t = s)
+    (u : OverArrows η t) : (u.map₂ f hst).val = F.map f.op u.val :=
+  rfl
+
+@[simp]
+lemma map₁_map₂ {F G : Cᵒᵖ ⥤ Type v} {η : F ⟶ A} {μ : G ⟶ A} (ε : F ⟶ G)
+    (hε : ε ≫ μ = η) {X Y : C} {s : yoneda.obj X ⟶ A} {t : yoneda.obj Y ⟶ A} (f : X ⟶ Y)
+    (hf : yoneda.map f ≫ t = s) (u : OverArrows η t) :
+    (u.map₁ ε hε).map₂ f hf = (u.map₂ f hf).map₁ ε hε :=
+  OverArrows.ext <| (elementwise_of% (ε.naturality f.op).symm) u.val
+
+/-- Construct an element of `OverArrows η s` with `F = yoneda.obj Y` from a suitable morphism
+    `f : X ⟶ Y`. -/
+def yonedaArrow {Y : C} {η : yoneda.obj Y ⟶ A} {X : C} {s : yoneda.obj X ⟶ A} (f : X ⟶ Y)
+    (hf : yoneda.map f ≫ η = s) : OverArrows η s :=
+  ⟨f, .of_yoneda_arrow hf⟩
+
+@[simp]
+lemma yonedaArrow_val {Y : C} {η : yoneda.obj Y ⟶ A} {X : C} {s : yoneda.obj X ⟶ A} {f : X ⟶ Y}
+    (hf : yoneda.map f ≫ η = s) : (yonedaArrow f hf).val = f :=
+  rfl
+
+/-- If `η` is also `yoneda`-costructured, then `OverArrows η s` is just morphisms of costructured
+    arrows. -/
+def costructuredArrowIso (s t : CostructuredArrow yoneda A) : OverArrows s.hom t.hom ≅ t ⟶ s where
+  hom p := CostructuredArrow.homMk p.val (by aesop_cat)
+  inv f := yonedaArrow f.left f.w
+
+end OverArrows
+
+/-- This is basically just `yoneda.obj η : (Over A)ᵒᵖ ⥤ Type (max u v)` restricted along the
+    forgetful functor `CostructuredArrow yoneda A ⥤ Over A`, but done in a way that we land in a
+    smaller universe. -/
+@[simps (config := { fullyApplied := false }) obj map]
```

Comment:
I am not sure we need this `fullyApplied := false`. The automatically generated simp lemmas look good to me (not just here, but throughout the file):
```suggestion
@[simps obj map]
```

Reply:
Thanks, I added these to work around some automation issues but it looks like it's no longer needed in the final version of the proof.